File: android/src/main/kotlin/com/pravera/flutter_activity_recognition/utils/ErrorHandleUtils.kt
package com.pravera.flutter_activity_recognition.utils

import com.pravera.flutter_activity_recognition.errors.ErrorCodes
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodChannel

/**
 * Utilities for error handling.
 *
 * @author Dev-hwang
 * @version 1.0
 */
class ErrorHandleUtils {
	companion object {
		/** Handles errors that occur in MethodChannel. */
		fun handleMethodCallError(result: MethodChannel.Result, errorCode: ErrorCodes) {
			result.error(errorCode.toString(), errorCode.message(), null)
		}

		/** Handles errors that occur in EventChannel. */
		fun handleMethodCallError(events: EventChannel.EventSink, errorCode: ErrorCodes) {
			events.error(errorCode.toString(), errorCode.message(), null)
		}
	}
}

File: ios/Classes/utils/ErrorHandleUtils.swift
//
//  ErrorHandleUtils.swift
//  flutter_activity_recognition
//
//  Created by Woo Jin Hwang on 9/5/24.
//

import Flutter
import Foundation

class ErrorHandleUtils {
  static func handleMethodCallError(result: FlutterResult, errorCode: ErrorCodes) {
    let error = FlutterError(code: errorCode.rawValue, message: errorCode.message(), details: nil)
    result(error)
  }
  
  static func handleStreamError(events: FlutterEventSink, errorCode: ErrorCodes) {
    let error = FlutterError(code: errorCode.rawValue, message: errorCode.message(), details: nil)
    events(error)
  }
}
